packages/app_center/lib/src/deb/local_deb_model.dart

```
@@ -0,0 +1,39 @@
+import 'package:app_center/packagekit.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:packagekit/packagekit.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:ubuntu_service/ubuntu_service.dart';
+
+part 'local_deb_model.freezed.dart';
+part 'local_deb_model.g.dart';
+
+@freezed
+class LocalDebData with _$LocalDebData {
+  factory LocalDebData({
+    required String path,
+    // TODO: use PackageKitPackageDetails type alias here (freezed seems to
+    //       expand typedefs, which causes some issues)
+    required PackageKitDetailsEvent? details,
+    int? activeTransactionId,
+  }) = _LocalDebData;
+}
+
+@riverpod
+class LocalDebModel extends _$LocalDebModel {
+  @override
+  Future<LocalDebData> build({required String path}) async {
+    final packageKit = getService<PackageKitService>();
+    final details = await packageKit.getDetailsLocal(path);
+    return LocalDebData(path: path, details: details);
+  }
+
+  Future<void> install() async {
+    if (!state.hasValue) return;
```

Do we want to throw an exception here, so that it comes back with an error?
I've been experimenting with having a `PrematureSnapOperationException` for these cases, but there are of course other ways to do it too.